### PR TITLE
Pass constraints to anchor point generator

### PR DESCRIPTION
**Pull request by LEleonora**
Currently the constraints are not being respected in the anchor point
generator. This could lead to infeasible points being selected as
anchor points. To fix this simply pass the commits to the design space
object that is used for creating the anchor points.

### CODE REVIEW

**Comment by javiergonzalezh** [comment]
Thanks for the PR! this is certainly a good catch!!! However, some of the tests are not passing yet. Could you please investigate that before we can merge? It would be great to add this to the library.

**Comment by LEleonora** [comment]
Hi, I checked that the results are feasible and everything is looking good at this end. Furthermore, I checked where the two code paths diverge.

The only part of the code where the modified space is being used is the call to initial_design and subsequently RandomDesign. There, a few infeasible points are being generated by the call to get_samples_without_constraints. In the patched version of the code, get_samples_with_constraints catches this and generates replacements. Naturally, this leads to different initial designs and a different random generator state down the road.

To sum up, it is probably best to replace the test case with the new output. I added a commit to do just that.

**Comment by apaleyes** [comment]
That looks good, let's merge.
